### PR TITLE
SPORTS-67 Add APIs to manage team, tournament category, team player and player.

### PR DESCRIPTION
**Pull request by sanjeevkpandit**
Teams
- `GET /api/v1/teams`
- `GET /api/v1/teams/:id`
- `POST /api/v1/teams`
- `PUT /api/v1/teams/:id`
- `DELETE /api/v1/teams/:id`

Players
- `GET /api/v1/players`
- `GET /api/v1/players/:id`
- `POST /api/v1/players`
- `PUT /api/v1/players/:id`
- `DELETE /api/v1/players/:id`

Team Players
- `GET /api/v1/team-players`
- `GET /api/v1/team-players/:id`
- `POST /api/v1/team-players`
- `PUT /api/v1/team-players/:id`
- `DELETE /api/v1/team-players/:id`

Tournament Categories
- `GET /api/v1/tournament-categories/:tournamentId`
- `POST /api/v1/tournament-categories`
- `PUT /api/v1/tournament-categories/:id`
- `DELETE /api/v1/tournament-categories/:id`

### CODE REVIEW

File: backend/src/constants/messages.ts [hunk 1]
```diff
@@ -52,3 +52,47 @@ export const categoryMessages = {
   unableToUpdate: 'Unable to update the category.',
   unableToCreate: 'Unable to create a new category.'
 };
+
+export const tournamentCategoryMessages = {
```
Comment (aviskarkc10) [inline backend/src/constants/messages.ts]: Why did we create this when categoryMessages was already present? Can't we just use that?

File: backend/src/services/tournamentCategoryService.ts [hunk 2]
```diff
@@ -0,0 +1,171 @@
...
+      throw new NotFoundError(tournamentMessages.notFound);
+    }
+
+    const tournamentCategories: Collection<TournamentCategory> = await new TournamentCategory({
+      tournament_id: tournamentId
+    }).fetchAll({withRelated: ['category', 'tournament']});
+
+    return {
+      data: tournamentCategories,
+      code: HttpStatus.OK,
```
Comment (aviskarkc10) [inline backend/src/services/tournamentCategoryService.ts]: length rule

File: backend/src/validators/paramsValidator.ts [hunk 3]
```diff
@@ -12,7 +12,7 @@ import { Request, Response, NextFunction } from 'express';
 export function validateURLParams(req: Request, res: Response, next: NextFunction) {
   const params = req.params;
 
-  if (isNaN(params.id)) {
+  if (params.id && isNaN(params.id)) {
```
Comment (aviskarkc10) [inline backend/src/validators/paramsValidator.ts]: ```params && params.id``` would be safer

File: backend/src/validators/paramsValidator.ts [hunk 4]
```diff
@@ -21,5 +21,14 @@ export function validateURLParams(req: Request, res: Response, next: NextFunctio
     });
   }
 
+  if (params.tournamentId && isNaN(params.tournamentId)) {
```
Comment (aviskarkc10) [inline backend/src/validators/paramsValidator.ts]: here too check for `params` first

File: backend/src/validators/paramsValidator.ts [hunk 5]
```diff
@@ -21,5 +21,14 @@ export function validateURLParams(req: Request, res: Response, next: NextFunctio
...
+    return res.status(HttpStatus.BAD_REQUEST).json({
+      error: {
+        code: HttpStatus.BAD_REQUEST,
+        message: 'Tournament ID must be a number.'
```
Comment (aviskarkc10) [inline backend/src/validators/paramsValidator.ts]: move this message to constants. And maybe the message could be 'Invalid tournament id'

File: backend/src/validators/playerValidator.ts [hunk 6]
```diff
@@ -0,0 +1,22 @@
...
+ * Validate player post/put object.
+ *
+ * @param req
+ * @param res
+ * @param next
+ */
+export function validatePlayerSchema(req: Request, res: Response, next: NextFunction) {
+  const result = validate(req.body, playerSchemas.playerSchema);
+
+  if (result.error) {
```
Comment (aviskarkc10) [inline backend/src/validators/playerValidator.ts]: `result && result.error`

File: backend/src/validators/teamPlayerValidator.ts [hunk 7]
```diff
@@ -0,0 +1,22 @@
...
+ * Validate player post/put object.
+ *
+ * @param req
+ * @param res
+ * @param next
+ */
+export function validateTeamSchema(req: Request, res: Response, next: NextFunction) {
+  const result = validate(req.body, teamPlayerSchemas.teamPlayerSchema);
+
+  if (result.error) {
```
Comment (aviskarkc10) [inline backend/src/validators/teamPlayerValidator.ts]: here too

File: backend/src/validators/teamValidator.ts [hunk 8]
```diff
@@ -0,0 +1,22 @@
...
+ * Validate player post/put object.
+ *
+ * @param req
+ * @param res
+ * @param next
+ */
+export function validateTeamSchema(req: Request, res: Response, next: NextFunction) {
+  const result = validate(req.body, teamSchemas.teamSchema);
+
+  if (result.error) {
```
Comment (aviskarkc10) [inline backend/src/validators/teamValidator.ts]: .

File: backend/src/validators/tournamentCategoryValidator.ts [hunk 9]
```diff
@@ -0,0 +1,22 @@
...
+ * Validate tournament category post object.
+ *
+ * @param req
+ * @param res
+ * @param next
+ */
+export function validateTournamentCategorySchema(req: Request, res: Response, next: NextFunction) {
+  const result = validate(req.body, tournamentCategorySchemas.tournamentCategorySchema);
+
+  if (result.error) {
```
Comment (aviskarkc10) [inline backend/src/validators/tournamentCategoryValidator.ts]: .

File: backend/src/constants/messages.ts (hunk 1 above)
Comment (sanjeevkpandit) [reply backend/src/constants/messages.ts]: `TournamentCategory` a different model.

Comment (aviskarkc10) [reply backend/src/constants/messages.ts]: What's the difference between `category` and `tournamentCategory`?

Comment (sanjeevkpandit) [reply backend/src/constants/messages.ts]: `TournamentCategory` is a linker table that sets a category with a tournament. For e.g. Men's Singles and Women's Singles categories can be associated with one specific tournament.

**Pull request merged**